src/main/java/com/mongodb/hibernate/jdbc/MongoConnectionProvider.java

```diff
@@ -0,0 +1,149 @@
+    private @Nullable ConnectionString connectionString;
+    private @Nullable String database;
+
+    // non-null after start() method is invoked successfully
+    private @Nullable MongoClient mongoClient;
+
+    private @Nullable String user;
+    private @Nullable String password;
+
+    @Override
+    public Connection getConnection() {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public void closeConnection(Connection connection) {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public boolean supportsAggressiveRelease() {
+        return false; // won't be used in container
+    }
+
+    @Override
+    public boolean isUnwrappableAs(Class<?> unwrapType) {
+        return ConnectionProvider.class.equals(unwrapType)
+                || MongoConnectionProvider.class.isAssignableFrom(unwrapType);
+    }
+
+    @Override
+    public <T> T unwrap(Class<T> unwrapType) {
+        if (isUnwrappableAs(unwrapType)) {
+            return unwrapType.cast(this);
+        } else {
+            throw new UnknownUnwrapTypeException(unwrapType);
+        }
+    }
+
+    @Override
+    public void configure(Map<String, Object> configurationValues) {
+        var jdbcUrl = ConfigurationHelper.getRequiredConfiguration(configurationValues, JAKARTA_JDBC_URL);
+        try {
+            this.connectionString = new ConnectionString(jdbcUrl);
+        } catch (IllegalArgumentException iae) {
+            throw new ConfigurationException(JAKARTA_JDBC_URL, "invalid MongoDB connection string", iae);
+        }
+        var database = this.connectionString.getDatabase();
+        if (database == null) {
+            throw new ConfigurationException(JAKARTA_JDBC_URL, "database must be provided");
+        }
+        this.database = database;
+        this.user = ConfigurationHelper.getOptionalConfiguration(configurationValues, JAKARTA_JDBC_USER);
+        this.password = ConfigurationHelper.getOptionalConfiguration(configurationValues, JAKARTA_JDBC_PASSWORD);
+    }
+
+    @Override
+    public void start() {
+        // connectionString and database are set as mandatory values in the above configure method
+        // if either is unset, exception would have been thrown and this method invocation would have be skipped
+        castNonNull(this.connectionString);
+        castNonNull(this.database);
+
+        var clientSettingsBuilder = MongoClientSettings.builder().applyConnectionString(connectionString);
+        if (this.user != null) {
+            var password = this.password == null ? null : this.password.toCharArray();
+            var credential = MongoCredential.createCredential(this.user, this.database, password);
+            clientSettingsBuilder.credential(credential);
+        }
+
+        var codecRegistry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry());
+        clientSettingsBuilder.codecRegistry(codecRegistry);
```

Comment:
not sure whether or how to set `codec`. Or `default` one suffices for now? Should we exclude the `POJO` codec?